File: api/models/move.go
package models

type Move struct {
	Id               int       `json:"id"`
	Name             string    `json:"name"`
	Accuracy         int       `json:"accuracy"`
	Pp               int       `json:"pp"`
	Priority         int       `json:"priority"`
	Power            int       `json:"power"`
	MoveType         MoveType  `json:"moveType"`
	MoveTypeID       uint      `json:"moveTypeID"`
	LearnedByPokemon []Pokemon `gorm:"many2many:pokemon_move_sets;" json:"learnedByPokemon"`
}

File: frontend/src/models/move.ts
import {MoveType, PokemonVariety} from "./pokemon.ts";


export interface Move {
    id: number;
    name: string;
    accuracy: number;
    pp: number;
    priority: number;
    power: number;
    moveType: MoveType;
    moveTypeID: number;
    learnedByPokemon: PokemonVariety[];
}